packages/auth/src/grant/routes.ts

```diff
@@ -179,6 +180,65 @@ function isContinuableGrant(grant: Grant): boolean {
   return !isRejectedGrant(grant) && !isRevokedGrant(grant)
 }
 
+function isGrantStillWaiting(grant: Grant): boolean {
+  if (!grant.wait) return false
+  const grantWaitTime = grant.createdAt.getTime() + grant.wait * 1000
+  const currentTime = Date.now()
+
+  return currentTime < grantWaitTime
+}
+
+async function pollGrantContinuation(
+  deps: ServiceDependencies,
+  ctx: ContinueContext,
+  continueId: string,
+  continueToken: string
+): Promise<void> {
+  const { config, grantService, accessService, accessTokenService } = deps
+
+  const grant = await grantService.getByContinue(continueId, continueToken)
+  if (!grant) {
+    ctx.throw(404, { error: 'unknown_request' })
+  }
+
+  if (isGrantStillWaiting(grant)) {
+    ctx.throw(401, { error: 'too_fast' })
+  }
+
+  /*
+    https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-15#name-continuing-during-pending-i
+    "When the client instance does not include a finish parameter, the client instance will often need to poll the AS until the RO has authorized the request."
+  */
+  if (grant.finishMethod) {
+    ctx.throw(401, { error: 'request_denied' })
+  } else if (
+    grant.state === GrantState.Pending ||
+    grant.state === GrantState.Processing
+  ) {
+    ctx.body = toOpenPaymentsGrantContinuation(grant, {
+      authServerUrl: config.authServerDomain
+    })
+    return
+  } else if (
+    grant.state !== GrantState.Approved ||
+    !isContinuableGrant(grant)
+  ) {
```

Comment:
If it's already a finalized grant, we should error always during continuation requests, correct? Just checking

Reply:
Yeah, that's correct. Finalized grants cannot be continued. 

https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-15#section-1.5
> Once in this state, no new access tokens can be issued, no subject information can be returned, and no interactions can take place. Once in this state, the grant request is dead and cannot be revived. If future access is desired by the client instance, a new grant request can be created, unrelated to this grant request.